src/engine/GraphStoreProtocol.h

```diff
@@ -94,43 +105,108 @@ class GraphStoreProtocol {
   }
-  FRIEND_TEST(GraphStoreProtocolTest, transformPost);
+  FRIEND_TEST(GraphStoreProtocolTest, transformPostAndTsop);
+
+  // Transform a SPARQL Graph Store Protocol TSOP to an equivalent ParsedQuery
+  // which is an SPARQL Update.
+  CPP_template_2(typename RequestT)(
+      requires ad_utility::httpUtils::HttpRequest<RequestT>) static ParsedQuery
+      transformTsop(const RequestT& rawRequest, const GraphOrDefault& graph) {
+    if (rawRequest.body().empty()) {
+      throw HttpError(boost::beast::http::status::no_content);
+    }
+    auto triples =
+        parseTriples(rawRequest.body(), extractMediatype(rawRequest));
+    auto convertedTriples = convertTriples(graph, std::move(triples));
+    updateClause::GraphUpdate up{{}, std::move(convertedTriples)};
+    ParsedQuery res;
+    res._clause = parsedQuery::UpdateClause{std::move(up)};
+    // Graph store protocol TSOP requests might have a very large body. Limit
+    // the length used for the string representation.
+    res._originalString =
+        absl::StrCat("Graph Store TSOP Operation\n",
+                     ad_utility::truncateOperationString(rawRequest.body()));
+    return res;
+  }
 
   // Transform a SPARQL Graph Store Protocol GET to an equivalent ParsedQuery
   // which is an SPARQL Query.
   static ParsedQuery transformGet(const GraphOrDefault& graph);
   FRIEND_TEST(GraphStoreProtocolTest, transformGet);
 
+  // Transform a SPARQL Graph Store Protocol PUT to equivalent ParsedQueries
+  // which are SPARQL Queries.
+  CPP_template_2(typename RequestT)(
+      requires ad_utility::httpUtils::HttpRequest<RequestT>) static std::
+      vector<ParsedQuery> transformPut(const RequestT& rawRequest,
+                                       const GraphOrDefault& graph) {
+    // TODO: The response codes are not conform to the specs. "If new RDF graph
+    //  content is created", then the status must be `201 Created`. "If
+    //  existing graph content is modified", then the status must be `200 OK`
+    //  or `204 No Content`.
+    auto triples =
+        parseTriples(rawRequest.body(), extractMediatype(rawRequest));
+    auto convertedTriples = convertTriples(graph, std::move(triples));
+    updateClause::GraphUpdate up{std::move(convertedTriples), {}};
+    ParsedQuery res;
+    res._clause = parsedQuery::UpdateClause{std::move(up)};
+    // Graph store protocol POST requests might have a very large body. Limit
+    // the length used for the string representation.
+    std::string stringRepresentation =
+        absl::StrCat("Graph Store PUT Operation\n",
+                     ad_utility::truncateOperationString(rawRequest.body()));
+    res._originalString = stringRepresentation;
+    std::string dropUpdate;
+    if (const auto* iri =
+            std::get_if<ad_utility::triple_component::Iri>(&graph)) {
+      dropUpdate =
+          absl::StrCat("DROP SILENT GRAPH ", iri->toStringRepresentation());
+    } else {
+      dropUpdate = "DROP SILENT DEFAULT";
+    }
```

Comment:
can be an immediately invoked lambda (more modern)

Reply:
And even more pressing:
Can't this simply call `transformDelete(theGraph)` , this looks like very much duplicated code from there at first glance.

Reply:
They differ in that one is `SILENT`. `SILENT` does not make a difference at the moment, but might in the future.